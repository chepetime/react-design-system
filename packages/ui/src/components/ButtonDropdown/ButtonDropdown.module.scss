@use "./../../styles" as styles;

:root {
  --buttondropdown-background-color: #{styles.$pink-500};
}

:global .dark {
  --buttondropdown-background-color: #{styles.$indigo-500};
}

.ButtonDropdown {
  @include styles.styles-reset;
  @include styles.text-wrapping;

  background: var(--buttondropdown-background-color);

  font-size: 1rem;

  @include styles.respond-to("md") {
    font-size: 1.5rem;
  }
  @include styles.respond-to("lg") {
    font-size: 2rem;
  }
  @include styles.feature-to("hover") {
    &:hover {
      background-color: pink;
    }
  }
}
