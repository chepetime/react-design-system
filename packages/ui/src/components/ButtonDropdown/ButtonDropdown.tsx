import * as React from "react";
import classnames from "classnames";
import css from "./ButtonDropdown.module.scss";

export interface ButtonDropdownProps {
  children?: React.ReactNode;
}

export const ButtonDropdown = React.forwardRef<HTMLDivElement, ButtonDropdownProps>(
  (props: ButtonDropdownProps, ref: React.Ref<HTMLDivElement>) => {
    return (
      <div className={classnames(css.ButtonDropdown)} ref={ref}>
        <p>ButtonDropdown component working!</p>
        <div>{props.children}</div>
      </div>
    );
  }
)

ButtonDropdown.displayName = "ButtonDropdown";
