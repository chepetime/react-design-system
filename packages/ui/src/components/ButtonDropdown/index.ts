export { ButtonDropdown } from "./ButtonDropdown";
export type { ButtonDropdownProps } from "./ButtonDropdown";