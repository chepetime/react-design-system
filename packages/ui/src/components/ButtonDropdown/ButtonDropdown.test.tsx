import React from "react";
import { render } from "@testing-library/react";
import { ButtonDropdown } from "./ButtonDropdown";

describe("ButtonDropdown", () => {
  test("renders the ButtonDropdown component", () => {
    render(<ButtonDropdown />);
  });
});
