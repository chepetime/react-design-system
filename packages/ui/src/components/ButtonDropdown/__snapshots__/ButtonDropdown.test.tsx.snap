// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonDropdown matches snapshot 1`] = `
<div>
  <div
    class="ButtonDropdown"
  >
    <p>
      ButtonDropdown component working!
    </p>
    <div />
  </div>
</div>
`;
